tools/cmake/mbed_greentea.cmake

```diff
@@ -70,6 +81,17 @@ macro(mbed_greentea_add_test)
 
     mbed_set_post_build(${MBED_GREENTEA_TEST_NAME})
 
+    add_test(
+        NAME ${MBED_GREENTEA_TEST_NAME}
+        COMMAND mbedhtrun
+            -m ${MBED_TARGET}
+            -f ${CMAKE_CURRENT_BINARY_DIR}/${MBED_GREENTEA_TEST_NAME}.bin
+            -p ${MBED_TARGET_SERIAL_PORT}
+            -t ${MBED_TARGET_SERIAL_NUM}
```

Comment:
It looks like `-m ${MBED_TARGET}` and `-t ${MBED_TARGET_SERIAL_NUM}` are optional too. Given serial port and mount point, it's usually enough for htrun to run a test on a local board. I think the target name `-m` is useful with the `--grm` option to let RaaS know which target to allocation, and the serial number might be to do with `--copy=COPY_METHOD` and `--reset=FORCED_RESET_TYPE` which might involve some DAPLink specifics (I'm not certain though).

The idea is it'd be good to make some of them optional so users don't need to specify them all.

Reply:
Anyway, this looks good as an initial PR. Maybe we can fine tune htrun command line options when working on #14908.

Reply:
The way CMake forwards optional arguments to CTest is through the [CMAKE_CTEST_ARGUMENTS](https://cmake.org/cmake/help/latest/variable/CMAKE_CTEST_ARGUMENTS.html) variable, which accepts a semicolon separated list of optional arguments. I was thinking it would be good to do something similar for `htrun`, i.e `MBED_HTRUN_OPTIONAL_ARGUMENTS`, and the user can pass any optional flags when they need to (whether that's automatically through use of `mbed-tools`, or using some other method). I agree it's probably better to implement the fine tuning of command line options in a follow up PR.

Reply:
Added `MBED_HTRUN_OPTIONAL_ARGUMENTS` and used it to make `HOST_TEST_DIRS` optional.

Reply:
Looks good to me